src/System.Text.Json/tests/NewtonTests/DateTimeConverterTests.cs

```diff
@@ -0,0 +1,164 @@
+            Assert.Equal(newOne.DateTimeField, c.DateTimeField);
+            Assert.Equal(newOne.DateTimeOffsetField, c.DateTimeOffsetField);
+            Assert.Equal(newOne.PostField, c.PostField);
+            Assert.Equal(newOne.PreField, c.PreField);
+
+            //test the other edge case too
+            c.DateTimeField = null;
+            c.DateTimeOffsetField = null;
+            c.PreField = "Pre";
+            c.PostField = "Post";
+            json = JsonSerializer.ToString(c);
+            Assert.Equal(@"{""PreField"":""Pre"",""DateTimeField"":null,""DateTimeOffsetField"":null,""PostField"":""Post""}", json);
+        }
+
+        [Fact]
+        public void SerializeUTC()
+        {
+            DateTimeTestClass c = new DateTimeTestClass();
+            c.DateTimeField = new DateTime(2008, 12, 12, 12, 12, 12, 0, DateTimeKind.Utc).ToLocalTime();
+            c.DateTimeOffsetField = new DateTime(2008, 12, 12, 12, 12, 12, 0, DateTimeKind.Utc).ToLocalTime();
+            c.PreField = "Pre";
+            c.PostField = "Post";
+            string json = JsonSerializer.ToString(c);
+            
+            NullableDateTimeTestClass newOne = JsonSerializer.Parse<NullableDateTimeTestClass>(json);
+            Assert.Equal(newOne.DateTimeField, c.DateTimeField);
+            Assert.Equal(newOne.DateTimeOffsetField, c.DateTimeOffsetField);
+            Assert.Equal(newOne.PostField, c.PostField);
+            Assert.Equal(newOne.PreField, c.PreField);
+
+            //test the other edge case too
+            c.DateTimeField = new DateTime(2008, 1, 1, 1, 1, 1, 0, DateTimeKind.Utc).ToLocalTime();
+            c.DateTimeOffsetField = new DateTime(2008, 1, 1, 1, 1, 1, 0, DateTimeKind.Utc).ToLocalTime();
+            c.PreField = "Pre";
+            c.PostField = "Post";
+            json = JsonSerializer.ToString(c);
+            
+            newOne = JsonSerializer.Parse<NullableDateTimeTestClass>(json);
+            Assert.Equal(newOne.DateTimeField, c.DateTimeField);
+            Assert.Equal(newOne.DateTimeOffsetField, c.DateTimeOffsetField);
+            Assert.Equal(newOne.PostField, c.PostField);
+            Assert.Equal(newOne.PreField, c.PreField);
+        }
+
+        [Fact]
+        public void BlogCodeSample()
+        {
+            PersonTest p = new PersonTest
+            {
+                Name = "Keith",
+                BirthDate = new DateTime(1980, 3, 8),
+                LastModified = new DateTime(2009, 4, 12, 20, 44, 55),
+            };
+
+            var options = new JsonSerializerOptions();
+            options.IgnoreNullValues = true;
+
+            string jsonText = JsonSerializer.ToString(p, new JsonSerializerOptions { IgnoreNullValues = true});
+
+            Assert.Equal(@"{""Name"":""Keith"",""BirthDate"":""1980-03-08T00:00:00"",""LastModified"":""2009-04-12T20:44:55""}", jsonText);
+        }
+    }
+
+    public class PersonTest
```

Comment:
Looks like this is a duplicate of the existing Person class. We should just re-use that one if possible.

Reply:
Good catch